### PR TITLE
Fix #76260 Add nullable annotation to INumberBase<>.TryParse

### LINKED ISSUE #76260: [API Proposal]: Add nullable annotation to INumberBase<>.TryParse
### Background and motivation

Similarly to #71309, would it be possible to add [MaybeNullWhen(false)] annotation to INumberBase<> interface?
currently we have 
``` c#
static abstract bool TryParse([NotNullWhen(true)] string? s, NumberStyles style, IFormatProvider? provider, out TSelf result);

static abstract bool TryParse(ReadOnlySpan<char> s, NumberStyles style, IFormatProvider? provider, out TSelf result);
```

### API Proposal

```csharp
static abstract bool TryParse([NotNullWhen(true)] string? s, NumberStyles style, IFormatProvider? provider, [MaybeNullWhen(false)] out TSelf result);

static abstract bool TryParse(ReadOnlySpan<char> s, NumberStyles style, IFormatProvider? provider, [MaybeNullWhen(false)] out TSelf result);
```

### API Usage

Like before but with nullability info when implemented by classes 

### Alternative Designs

_No response_

### Risks

_No response_

### PR DESCRIPTION
**Pull request by MichalBrylka**
Fix #76260

### CODE REVIEW

File: src/libraries/System.Private.CoreLib/src/System/Numerics/INumberBase.cs [hunk 1]
```diff
@@ -346,6 +346,6 @@ static virtual TSelf CreateTruncating<TOther>(TOther value)
         /// <param name="result">On return, contains the result of successfully parsing <paramref name="s" /> or an undefined value on failure.</param>
         /// <returns><c>true</c> if <paramref name="s" /> was successfully parsed; otherwise, <c>false</c>.</returns>
         /// <exception cref="ArgumentException"><paramref name="style" /> is not a supported <see cref="NumberStyles" /> value.</exception>
-        static abstract bool TryParse(ReadOnlySpan<char> s, NumberStyles style, IFormatProvider? provider, out TSelf result);
+        static abstract bool TryParse(ReadOnlySpan<char> s, NumberStyles style, IFormatProvider? provider, [MaybeNullWhen(false)] out TSelf result);
```
Comment (stephentoub) [inline src/libraries/System.Private.CoreLib/src/System/Numerics/INumberBase.cs:349]: Thanks.  You'll also need to attribute these methods in the ref assembly:
https://github.com/dotnet/runtime/blob/main/src/libraries/System.Runtime/ref/System.Runtime.cs

Comment (MichalBrylka) [reply src/libraries/System.Private.CoreLib/src/System/Numerics/INumberBase.cs:349]: Thanks. I did that just after my first commit

**Comment by dakersnar** [comment]
This needs to be in for 7.0, right?

**Comment by MichalBrylka** [comment]
@dakersnar I'm not sure if you ask me but it would be awesome if it would go to 7.0.0. Risk are very slim that it will blow anything

**Review by tannergooding — APPROVED** [review]

**Pull request merged**